service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java

```diff
@@ -563,6 +578,43 @@ void shouldReturnFalseWhenNoFutureHearingExists() {
         assertThat(hearingBookingInFuture).isFalse();
     }
 
+    @Nested
+    class GetNextHearingAfterCmo {
+        @Test
+        void shouldReturnExpectedNextHearingBooking() {
+            List<Element<HearingBooking>> hearingBookings = createHearingBookingElements();
+            HearingBooking issueResolutionHearing = createHearingBooking(futureDate.plusDays(6),
+                futureDate.plusDays(7), ISSUE_RESOLUTION, UUID.randomUUID());
+            hearingBookings.add(element(issueResolutionHearing));
+
+            CaseData caseData = CaseData.builder().hearingDetails(hearingBookings).build();
+            Optional<HearingBooking> nextHearingBooking = caseData.getNextHearingAfterCmo(cmoID);
```

Comment:
To be honest I would consider preparing the data here, the hearings are such fundamentals to the functionality is difficult to trace what's happening in this test.One hearing is added here, so it's confusing why this hearing is special, I would consider preping the data at the beginning of that test. 

The second thing I would recommend using different cmoId, to be honest I don't know why this even works now, as all hearings on the list have got the same cmoId. They should have different ids and we should explicitly target findings.